Add docs for gateway visibility (#9022)
wso2/docs-apim: en/docs/administer/multiple-gateways/configure-gateway-visibility.md

# Configure Gateway visibility

> Access to Gateways can be restricted based on user roles either from the admin portal or via the `deployment.toml` file.

Reviewer: ```suggestion
Access to Gateways can be restricted based on user roles either from the Admin Portal or via the `<API-M_HOME>/repository/conf/deployment.toml` file.
```

> When the permission type for the role-based restriction is set to ALLOW, only users with the specified roles are permitted to use the gateway environment at the dev portal.

Reviewer: ```suggestion
When the permission type for the role-based restriction is set to **ALLOW**, only users with the specified roles are permitted to use the gateway environment at the Developer Portal.
```

> When the permission type for the role-based restriction is set to DENY, users with the specified roles are not allowed to use the gateway environment at the dev portal.

Reviewer: ```suggestion
Similarly, when the permission type for the role-based restriction is set to **DENY**, users with the specified roles are not allowed to use the gateway environment at the Developer Portal.
```

!!! note 

    - When implementing role-based access restrictions, it is important for administrators to exercise caution and be mindful of potential scenarios where users, may not have visibility or access to gateway environments. Kindly be cautious when restricting gateways.
    - A Gateway can possess either a whitelist or a blacklist, but not both simultaneously.

## Option 1: Configure visibility for a Gateway environment via the Admin Portal

1. [Start WSO2 API Manager]({{base_path}}/install-and-setup/install/installing-the-product/running-the-api-m/#starting-the-server).

   This will start WSO2 API Manager in the all-in-one mode, which includes the default Gateway as well.

2.  Sign in to the Admin Portal.

    `https://<hostname>:9443/admin`

    Example: `https://localhost:9443/admin`

    Let's use `admin` as your username and password to sign in.

3.  Add a new Gateway Environment.

   1. Click **Gateways**, and then click **Add Gateway Environment**.

      [![Menu to add Gateway environment]({{base_path}}/assets/img/deploy/add-gateway-environment-menu.png)]({{base_path}}/assets/img/deploy/add-gateway-environment-menu.png)

>    2. Enter a name, display name, description, visibility and a virtual host.

Reviewer: ```suggestion
   2. Enter a name, display name, description, visibility and a virtual host as follows.
```

      The virtual hosts will define each of the custom hostnames. It is mandatory to specify a VHost when you create a Gateway environment.

      | **Environment** | **Display Name** | **Description**                               | **Visibility** | **Virtual Host** |
      |-------------|--------------|----------------|--------------|--------------|
      | us-region   | US Region    | Gateway environment deployed in the US region. | Public         | us.wso2.com    |

      [![Menu to add Gateway environment visibility]({{base_path}}/assets/img/deploy/add-gateway-visibility-menu.png)]({{base_path}}/assets/img/deploy/add-gateway-visibility-menu.png)

>       The visibility of a Gateway can be set by selecting the permission type. The list of allowed/denied roles should be provided for the ALLOW of DENY types.

Reviewer: We might be able to add lines 44 - 48 into a note section. WDYT?

Author: +1
